src/contracts/ChugSplash.sol

```diff
@@ -7,6 +7,20 @@ import "lib/solidity-stringutils/strings.sol";
 
 contract ChugSplash is Script, Test {
     using strings for *;
+
+    string constant NONE = "none";
+
+    // Required env vars
+    string privateKey = vm.envString("PRIVATE_KEY");
+    string network = vm.envString("NETWORK");
+
+    // Optional env vars
+    bool silent = vm.envOr("SILENT", false);
+    address newOwnerAddress = vm.envOr("NEW_OWNER", vm.addr(vm.envUint("PRIVATE_KEY")));
+    string newOwner = vm.toString(newOwnerAddress);
+    bool withdrawFunds = vm.envOr("WITHDRAW_FUNDS", true);
+    string ipfsUrl = vm.envOr("IPFS_URL", NONE);
+    bool skipStorageCheck = vm.envOr("SKIP_STORAGE_CHECK", false);
```

Comment:
So I think this is better UX than passing in all of these via parameters. However, there are some issues with this I want to point out:

1. Not all of this really fits logically into a .env file. Environment variables cannot (or should not) be saved to version control. They should generally be used to store values that are unique to the environment and/or secret. Not all of these values necessarily match those two features.
2. If these values are configured via environment variables, they cannot be changed while the script is running. So for example, if you wanted to silence one function but not a different one, you wouldn't be able to do that. If you wanted to perform two deployments against two different networks in a single script, you wouldn't be able to do that either. Not all of these values necessarily need to be this flexible, but I'm guessing some probably should be.
3. We're spreading configuration parameters across multiple files. While this is better than doing the configuration via function parameters, it's not ideal IMO. 

Here's what I recommend for a future iteration: 
- We should think through each parameter we want to be configured via an external file and make sure that it makes sure there aren't strong reasons why users might want to be able to configure these parameters differently for different function calls. 
- We should move as many of these configuration parameters as possible into either the foundry.toml file or the ChugSplash config file so that they can be included in version control. I think we should reserve environment variables for things that are actually secret like private keys, and API access credentials.

Reply:
Yeah totally agree, thanks for making a note of this!